cpp/src/arrow/util/async_generator.h

```diff
@@ -1247,6 +1248,8 @@ class BackgroundGenerator {
   }
 
  protected:
+  static constexpr uint64_t kUnlikelyThreadId{std::numeric_limits<uint64_t>::max()};
```

Comment:
I couldn't see any reason this wouldn't work but why not just use the `uint64_t` version of `std::thread::id()`?  Is it to allow this to be `constexpr`?

Reply:
Hmm, I hadn't thought of that. But it seems simpler to use a hardcoded number like this.